staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,46 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	WorkEstimator WorkEstimatorConfiguration
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimator holds work estimator parameters related to list requests.
+	ListWorkEstimator ListWorkEstimatorConfiguration
+	// MutatingWorkEstimator holds work estimator parameters
+	// related to requests that mutate watched objects.
+	MutatingWorkEstimator MutatingWorkEstimatorConfiguration
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	MaximumSeats uint64
```

Comment:
What happens if this is too low or too high?

What happens if e.g. we just set this to all of the seats available to the PL?

What if this is greater than the number of seats available to the PL?

How is the person configuring supposed to know what the range is? Should they look at shares and do the math with the configured global concurrency level?